lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -4,9 +4,31 @@ module ActiveModel
   class Serializer
     class Adapter
       class JsonApi < Adapter
+        @root = :data
+        DEFAULT_ATTRIBUTES = [:type, :id]
+
+        def self.params(permitted, associations)
+          relationships = {}
+          associations.each do |assoc|
+             relationships[assoc] = {}
+             relationships[assoc][@root] = DEFAULT_ATTRIBUTES
+           end
+
+          return :type, {attributes: permitted}, {relationships: relationships}
+        end
+
+        def self.parse(params)
+          attrs, assoc = {}, {}
+          attrs = params['attributes'] if params['attributes']
+          assoc = params['relationships'].map {|a| {a.shift => a.first['data']['type'].camelize.singularize.safe_constantize.find(a.shift['data']['id'])}} if params['relationships']
```

Comment:
The `.find()` call here is actually not needed, as we can get away with just the IDs. I think something along the lines of: https://gist.github.com/beauby/6e6fa23d09aa0745ad58 would give an ActiveRecord-friendly payload, while not relying explicitly on it.